### DISCUSSION TITLE
Provide MySQL 8 Support

### DISCUSSION THREAD

**Discussion by waza-ari** [directus/directus#2795] [Archive]
It would be nice to support MySQL 8 including the more recent default authentication plugin caching_sha2_password out of the box without the need to fall back to native password authentication

**Comment by WoLfulus** [comment]
since it's a change in the protocol, the native connector (pdo) should support it first. seems like the recommended way atm is to fallback to the old method.

see notes regarding MySQL 8 here:

https://www.php.net/manual/en/ref.pdo-mysql.php

**Comment by waza-ari** [comment]
Hey, thank you for the answer! It basically says:

"When running a PHP version before 7.1.16, or PHP 7.2 before 7.2.4, set MySQL 8 Server's default password plugin to mysql_native_password or else you will see errors similar to The server requested authentication method unknown to the client [caching_sha2_password] even when caching_sha2_password is not used."

For me this suggests that the only thing we need to do is to update the PHP version? I'm using the official docker image, so maybe its just about bumping the PHP version?

Edit:

Looks like it is 7.3 already

```
root@673ae257e0a7:/var/directus# php -v
PHP 7.3.19 (cli) (built: Jun 11 2020 19:59:27) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.19, Copyright (c) 1998-2018 Zend Technologies
```

Not completely sure where the error comes from then

**Comment by WoLfulus** [comment]
Oh sorry, I might have assumed you were using a non supported version in that case. 

Anyways, I think it's not supported yet on 7.3 versions since I was able to find some reports and docs are kinda misleading to the actual status. You might wanna try 7.4 instead though since it seems to have landed there.

Let me know how if that works, since it doesn't seem to need any changes from our side.
